lib/beaker/test_suite.rb

```diff
@@ -172,7 +172,9 @@ def write_junit_xml(xml_file)
             end
             suite.add_child(properties)
 
-            @test_cases.each do |test|
+            test_cases_to_report = @test_cases
+            test_cases_to_report = @test_cases.sort { |x,y| y.runtime <=> x.runtime } if time_sort
```

Comment:
Oh, I was thinking that `@test_cases.sort` was an in-place sort but it actually returns a new sorted array. (Basically I was confusing Ruby's `Array.sort` with `Array.sort!`)